src/iop/tonecurve.c

```diff
@@ -1200,17 +1533,46 @@ static gboolean dt_iop_tonecurve_draw(GtkWidget *widget, cairo_t *crf, gpointer
         pango_font_description_set_absolute_size(desc, PANGO_SCALE);
         layout = pango_cairo_create_layout(cr);
         pango_layout_set_font_description(layout, desc);
+
+        float Y_picker_min = 0.0f, Y_picker_max = 0.0f, Y_picker_mean = 0.0f;
+
+        // convert the picker samples to XYZ to show the real Y luminance value if we work in XYZ or RGB mode
+        if (autoscale_ab == s_scale_automatic_xyz || autoscale_ab == s_scale_automatic_rgb)
+        {
+          float XYZ[3];
+
+          dt_Lab_to_XYZ((const float*)&raw_min[ch_L], XYZ);
+          Y_picker_min = XYZ[1];
+
+          dt_Lab_to_XYZ((const float*)&raw_max[ch_L], XYZ);
+          Y_picker_max = XYZ[1];
+
+          dt_Lab_to_XYZ((const float*)&raw_mean[ch_L], XYZ);
+          Y_picker_mean = XYZ[1];
+        }
+
         picker_scale(raw_mean, picker_mean);
         picker_scale(raw_min, picker_min);
         picker_scale(raw_max, picker_max);
 
+        if (autoscale_ab == s_scale_automatic_xyz || autoscale_ab == s_scale_automatic_rgb)
+        {
+          raw_min[ch_L] = Y_picker_min;
```

Comment:
isn't here picker_* expected instead of the raw_*?

Reply:
picker_* are the global pickers you set in the left panels. raw_* are the local ones you set in the module. We take care of them lines 1482-1503